[ MB-16421 ] Adding DOSFSC pricer

rogeruiz opened the pull request:
## [Jira ticket](https://dp3.atlassian.net/browse/MB-16421)

## Summary

- [x] Adding interface and mocks for DOSFSC
- [x] Adding in the service for DOSFSC;
- [x] Adding DOSFSC to MTO service item updater

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1. Access the Office app
2. Create a testharness `HHGMoveInSITEndsYesterday` move
3. Login as the prime
4. Create a payment request for Domestic Origin SIT FSC

robot-mymove commented:
<!--
  0 failure: 
  0 warning: 
  1 messages
  
  DangerID: danger-id-Danger;
-->



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>:link: <a href="https://dp3.atlassian.net/browse/MB-16421">MB-16421</a></td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d961ee755974d738a4ea48657b5b11fd9b60b235
</p>

mlm55 commented on pkg/testdatagen/scenario/shared.go:
```diff
@@ -4154,9 +4154,6 @@ func createHHGWithOriginSITServiceItems(
 		}
-		if serviceItem.ReService.Code != models.ReServiceCodeDOSFSC {
-			paymentServiceItems = append(paymentServiceItems, paymentItem)
-		}
```
I think we just need to remove the if statement around `paymentServiceItems = append(paymentServiceItems, paymentItem)` - we still want to create the payment requests

rogeruiz replied:
oh 🤦🏽 thank you.

mlm55 commented on pkg/services/mto_service_item/mto_service_item_updater.go:
```diff
@@ -136,9 +136,11 @@ func (p *mtoServiceItemUpdater) updateServiceItem(appCtx appcontext.AppContext,
+		// only updates the address fields if the service item is of DDDSIT,
+		// DOSFSC, DDSFSC.
 		if (serviceItem.ReService.Code == models.ReServiceCodeDDDSIT ||
+			serviceItem.ReService.Code == models.ReServiceCodeDOSFSC ||
```
question: Should this be here? looks like this is for setting the destination sit final address rather than origin sit final address

rogeruiz replied:
Ah thank you. good call. I'll revert this.

mlm55 commented on pkg/services/ghcrateengine/domestic_origin_sit_fuel_surcharge_pricer_test.go:
Suggestion: i think there may be some lines of code that don't actually affect the tests - might be worth looking at `domestic_destination_sit_fuel_surcharge_pricer_test.go`?

rogeruiz replied:
Could you say a bit more here? This was mostly taken from the DDSFSC test file, but it should now be testing for DOSFSC.

mlm55 replied:
One example is that removing lines 66-73 doesn't seem to affect the tests. I'm not sure of the history here/why they were added originally, but they don't seem to be adding anything at the moment.

robot-mymove commented:
# Test coverage on server failed.
  **Last run: Tue 25 Jul 2023 08:23:16 PM UTC**
  Refer to [this report](https://output.circle-artifacts.com/output/job/7bc476f1-6c17-497b-a5ec-93a1ff5f452f/artifacts/0/test-results/gotest/app/go-coverage.html) to see coverage details for the server app.
  Documentation for test coverage can be found [at this link](https://transcom.github.io/mymove-docs/docs/about/development/test-coverage-reports).

mlm55 commented on pkg/services/mto_service_item/mto_service_item_updater.go:
```diff
@@ -136,8 +136,9 @@ func (p *mtoServiceItemUpdater) updateServiceItem(appCtx appcontext.AppContext,
-		// address fields if the service item is of DDDSIT.
+		// expression looks for the DDDSIT or DDSFSC service codes and only
+		// updates the address fields if the service item is of DDDSIT or
+		// DDSFSC.
```
praise: thanks for updating this comment

robot-mymove commented:
# Test coverage on client failed.
  **Last run: Thu 27 Jul 2023 02:20:53 AM UTC**
  Refer to [this report](https://output.circle-artifacts.com/output/job/93fafeca-5d8a-4710-8b9d-abbea8faafdc/artifacts/0/coverage/lcov-report/index.html) to see coverage details for the client app.
  Documentation for test coverage can be found [at this link](https://transcom.github.io/mymove-docs/docs/about/development/test-coverage-reports).

rogeruiz commented:
Alrighty, this should now be working. It was verified by @mlm55 that it worked and then I went ahead and fixed the tests to get it the CI passing and addressed any merge conflicts from the default branch.
